packages/ifpack2/src/Ifpack2_BlockRelaxation_def.hpp

```diff
@@ -151,7 +151,7 @@ getValidParameters () const
 
   validParams->set("partitioner: line detection threshold",(double)0.0);
   validParams->set("partitioner: PDE equations",(int)1);
-  Teuchos::RCP<Tpetra::MultiVector<typename MatrixType::scalar_type,
+  Teuchos::RCP<Tpetra::MultiVector<double,
```

Comment:
Are coordinates always double?  Could they perhaps be float?

Reply:
We have had a discussion around that issue in MueLu, on possible option to fix this would be to create a new template type called `Real` that would be essentially the same as `Scalar` but forbidding complex. I guess we could also manually template on more types but that would lead to some cumbersome checks when pulling data off the parameter list.

Overall we decided that coordinates will always be `double` for the time being so the above code should be fine.